### ISSUE TITLE
pod install fail on RN 0.72.x & New Arch

### ISSUE THREAD

**Issue by retyui** [react-native-async-storage/async-storage#998]
### What happened?

`pod install` fail with the next error on RN+IOS+New_Arch

```java
[Codegen] Done.
Updating local specs repositories
Analyzing dependencies
[!] CocoaPods could not find compatible versions for pod "RNCAsyncStorage":
  In Podfile:
    RNCAsyncStorage (from `../node_modules/@react-native-async-storage/async-storage`)

Specs satisfying the `RNCAsyncStorage (from `../node_modules/@react-native-async-storage/async-storage`)` dependency were found, but they required a higher minimum deployment target.
Couldn't install Pods. Updating the Pods project and trying again...
Command `pod install` failed.
└─ Cause: CocoaPods could not find compatible versions for pod "RNCAsyncStorage":
  In Podfile:
    RNCAsyncStorage (from `../node_modules/@react-native-async-storage/async-storage`)

Specs satisfying the `RNCAsyncStorage (from `../node_modules/@react-native-async-storage/async-storage`)` dependency were found, but they required a higher minimum deployment target.

error Command failed with exit code 1.

```

### Version

1.19.1

### What platforms are you seeing this issue on?

- [ ] Android
- [X] iOS
- [ ] macOS
- [ ] Windows
- [ ] web

### System Information

```true
System:
  OS: macOS 13.5
  CPU: (10) arm64 Apple M1 Pro
  Memory: 7.56 GB / 32.00 GB
  Shell:
    version: 3.6.1
    path: /opt/homebrew/bin/fish
Binaries:
  Node:
    version: 18.17.0
    path: /usr/local/bin/node
  Yarn:
    version: 1.22.19
    path: /opt/homebrew/bin/yarn
  npm:
    version: 9.7.2
    path: /opt/homebrew/bin/npm
  Watchman:
    version: 2023.07.10.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.12.1
    path: /opt/homebrew/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 22.4
      - iOS 16.4
      - macOS 13.3
      - tvOS 16.4
      - watchOS 9.4
  Android SDK: Not Found
IDEs:
  Android Studio: Flamingo 2022.2.1 Patch 2 Flamingo 2022.2.1 Patch 2
  Xcode:
    version: 14.3.1/14E300c
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 11.0.19
    path: /usr/bin/javac
  Ruby:
    version: 2.6.10
    path: /usr/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.72.3
    wanted: 0.72.3
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: true
  newArchEnabled: true
```

### Steps to Reproduce

```
cd MyRn72
yarn add @react-native-async-storage/async-storage
RCT_NEW_ARCH_ENABLED=1 npx pod-install
```

**Comment by retyui** [comment]
test on RN 0.72.x, the same issue

**Comment by retyui** [comment]
@tido64 no it's a different issue

**Comment by tido64** [comment]
Do you have a repro? The error message indicates that you're not targeting a high enough iOS version.

**Comment by retyui** [comment]
fixed by defining a `13.4` platform version in `ios/Podfile`

```rb
platform :ios, "13.4"
```
